src/Microsoft.IdentityModel.Tokens/AppContextSwitches.cs

```diff
@@ -16,6 +16,42 @@ internal static class AppContextSwitches
         /// </summary>
         internal const string UseClaimsIdentityTypeSwitch = "Microsoft.IdentityModel.Tokens.UseClaimsIdentityType";
 
-        internal static bool UseClaimsIdentityType() => (AppContext.TryGetSwitch(UseClaimsIdentityTypeSwitch, out bool useClaimsIdentityType) && useClaimsIdentityType);
+        private static bool? _useClaimsIdentity;
+
+        internal static bool UseClaimsIdentityType => _useClaimsIdentity ??= (AppContext.TryGetSwitch(UseClaimsIdentityTypeSwitch, out bool useClaimsIdentityType) && useClaimsIdentityType);
```

Comment:
I wonder if we should use a simpler pattern here to ensure that tiered JIT can recognize and elide the branches. In general, if the tiered JIT can see that a static bool will always be `true` or `false`, it can take out branches resulting in faster code. I'm not sure if the JIT can recognize this pattern with a nullable boolean as immutable, since the field isn't readonly.

One pattern looks like:

```csharp
 internal static bool UseClaimsIdentityType { get; } = AppContext.TryGetSwitch(UseClaimsIdentityTypeSwitch, out bool useClaimsIdentityType) && useClaimsIdentityType;
 ```

This makes a readonly field and a readonly property that simply returns the field value. The initialization happens in a static constructor, and after that the value will never change. So if the code gets re-JIT'd, the JIT will recognize the value as always `true` or `false`.

See https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-8/#tiering-and-dynamic-pgo for a good description of how tiering works.

Reply:
How can this approach be used in tests? Currently, we reset `_useClaimsIdentity`, so it is refreshes from the switch value.

Reply:
In dotnet/runtime we use something called the [RemoteExecutor](https://github.com/dotnet/arcade/tree/main/src/Microsoft.DotNet.RemoteExecutor) which spins up a new process with the desired AppContext switches, and then tests inside that process.

Example: https://github.com/dotnet/runtime/blob/4f38f921ddeb6773d1e8316298215d84dc643cdd/src/libraries/System.Threading.ThreadPool/tests/ThreadPoolTests.cs#L198-L211